numpy/core/tests/test_custom_dtypes.py

```diff
@@ -51,6 +51,12 @@ def test_repr(self):
     def test_dtype_name(self):
         assert SF(1.).name == "_ScaledFloatTestDType64"
 
+    def test_sfloat_structured_dtype_printing(self):
+        dt = np.dtype([("id", int), ("value", SF(0.5))])
+        # repr of structured dtypes need special handling because the
+        # implementation bypasses the object repr
+        assert "('value', '_ScaledFloatTestDType64')" in repr(dt)
```

Comment:
A bit confusing, I woul;d have thout if the test above does `SF(1.).name == "_ScaledFloatTestDType64"` we get `_ScaledFloatTestDType6464` here :).